store/errors.go

```diff
@@ -5,8 +5,34 @@ import (
 	"net/http"
 
 	"github.com/xmidt-org/argus/model"
+	"github.com/xmidt-org/httpaux"
 )
 
+type sanitizedError interface {
+	// Sanitized returns an HTTP error with content
+	// that should be safe to share across API boundaries.
+	// This provides a mechanism to prevent leaking
+	// sensitive error data coming from the datastores.
+	Sanitized() error
+}
+
+type SanitizedError struct {
+	Err          error
+	SanitizedErr httpaux.Error
```

Comment:
Running code similar to yours gives us the `httpaux.Error`'s status code:
```
func getCode(err error) int {
  var s SanitizedError
  if errors.As(err, &s) {
    err = s.Sanitized()
  }
  var c kithttp.StatusCoder
  if errors.As(err, &c) {
    return c.StatusCode()
  }
  return 0
}
```
While this code will give us a status code from the other error (and most likely no status code at all):
```
func getCode(err error) int {
  var c kithttp.StatusCoder
  if errors.As(err, &c) {
    return c.StatusCode()
  }
  return 0
```

My suggestion is either:
- embed the `httpaux.Error` struct, so SanitizedError immediately has `StatusCode()` and `Headers()` (which has a similar issue).  You do get some other functions too though, like `MarshalJSON()`.
- implement a `StatusCode()` and `Headers()` function for `SanitizedError` that return the values from the `httpaux.Error`.  This is good if you don't want the other functions.